cvxpy/tests/test_KKT.py

```diff
@@ -0,0 +1,301 @@
+        ]
+        var_pairs = [(X, np.array([[ 4.00000001,  4.        , -1.05467058, -1.05467058],
+                                   [ 4.        ,  4.00000001, -1.05467058, -1.05467058],
+                                   [-1.05467058, -1.05467058,  0.27941584,  0.27674984],
+                                   [-1.05467058, -1.05467058,  0.27674984,  0.27941584]]))]
+        obj_pair = (obj, 5.422574709567284)
+        sth = STH.SolverTestHelper(obj_pair, var_pairs, con_pairs)
+        return sth
+
+    @staticmethod
+    def tf_3() -> STH.SolverTestHelper:
+        """
+        Tests nonneg flag
+        """
+        X = cp.Variable(shape=(4,4), nonneg=True)
+        obj = cp.Minimize(cp.log_sum_exp(X))
+        cons = [cp.norm2(X) <= 10, X[0][1] >= 4, X[0][1] <= 8]
+        con_pairs = [
+            (cons[0], None),
+            (cons[1], None),
+            (cons[2], None),
+        ]
+        var_pairs = [(X, np.array([[1.19672119e-07, 4.00000000e+00, 1.19672119e-07, 1.19672119e-07],
+                                   [8.81309115e-08, 1.19672119e-07, 8.81309115e-08, 8.81309115e-08],
+                                   [8.81309115e-08, 1.19672119e-07, 8.81309115e-08, 8.81309115e-08],
+                                   [8.81309115e-08, 1.19672119e-07, 8.81309115e-08, 8.81309088e-08]]
+                                  ))]
+        obj_pair = (obj, 4.242738008082711)
+        sth = STH.SolverTestHelper(obj_pair, var_pairs, con_pairs)
+        return sth
+
+    @staticmethod
+    def tf_4() -> STH.SolverTestHelper:
+        """
+        Tests nonpos flag
+        """
+        X = cp.Variable(shape=(3, 3), nonpos=True)
+        obj = cp.Minimize(cp.norm2(X))
+        cons = [cp.log_sum_exp(X) <= 2, cp.sum_smallest(X, 5) >= -10]
+        con_pairs = [
+            (cons[0], None),
+            (cons[1], None),
+        ]
+        var_pairs = [(X, np.array([[-0.19722456, -0.19722456, -0.19722456],
+                                   [-0.19722456, -0.19722456, -0.19722456],
+                                   [-0.19722456, -0.19722456, -0.19722456]]))]
+        obj_pair = (obj, 0.5916736882031846)
+        sth = STH.SolverTestHelper(obj_pair, var_pairs, con_pairs)
+        return sth
+
+    def test_tf_1(self, places: int = 4):
+        sth = StandardTestFlags.tf_1()
+        sth.solve(solver='SCS')
+        sth.verify_objective(places)
+        sth.check_primal_feasibility(places)
+        sth.verify_primal_values(places)
+        sth.check_complementarity(places)
+        sth.verify_dual_values(places)
+        sth.check_stationary_lagrangian(places)
+
+    def test_tf_2(self, places: int = 2):
+        sth = StandardTestFlags.tf_2()
+        sth.solve(solver='SCS')
+        sth.verify_objective(places)
+        sth.check_primal_feasibility(places)
+        sth.verify_primal_values(places)
+        sth.check_complementarity(places)
+        sth.verify_dual_values(places)
+        sth.check_stationary_lagrangian(places=4)
```

Comment:
Is there a reason why you hard-code ``places=4`` here?

Reply:
Yeah, some of the other test could only verify correctness till the 2nd decimal place --- but I wanted to check if the stationarity function could give higher accuracy. I'll remove this